test/nuxt/use-async-data.test.ts

```diff
@@ -1153,4 +1153,22 @@ describe('useAsyncData', () => {
 
     vi.useRealTimers()
   })
+
```

Comment:
This test does not detect the bug behaviour. Maybe we still need it 😅

Reply:
pushed a test that seems to catch it!